pkg/tbtc/node.go

```diff
@@ -205,3 +268,136 @@ func (n *node) joinDKGIfEligible(seed *big.Int, startBlockNumber uint64) {
 		logger.Infof("not eligible for DKG with seed [0x%x]", seed)
 	}
 }
+
+// decideMemberFate decides what the member will do in case it failed to publish
+// its DKG result. Member can stay in the group if it supports the same group
+// public key as the one registered on-chain and the member is not considered as
+// misbehaving by the group.
+func (n *node) decideMemberFate(
+	memberIndex group.MemberIndex,
+	dkgResultChannel chan *DKGResultSubmittedEvent,
+	publicationStartBlock uint64,
+	result *dkg.Result,
+) ([]group.MemberIndex, error) {
+	dkgResultEvent, err := n.waitForDkgResultEvent(
+		dkgResultChannel,
+		publicationStartBlock,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	groupPublicKeyBytes, err := result.GroupPublicKeyBytes()
+	if err != nil {
+		return nil, err
+	}
+
+	// If member doesn't support the same group public key, it could not stay
+	// in the group.
+	if !bytes.Equal(groupPublicKeyBytes, dkgResultEvent.GroupPublicKeyBytes) {
+		return nil, fmt.Errorf(
+			"[member:%v] could not stay in the group because "+
+				"the member do not support the same group public key",
+			memberIndex,
+		)
+	}
+
+	misbehavedSet := make(map[group.MemberIndex]struct{})
+	for _, misbehavedID := range dkgResultEvent.Misbehaved {
+		misbehavedSet[misbehavedID] = struct{}{}
+	}
+
+	// If member is considered as misbehaved, it could not stay in the group.
+	if _, isMisbehaved := misbehavedSet[memberIndex]; isMisbehaved {
+		return nil, fmt.Errorf(
+			"[member:%v] could not stay in the group because "+
+				"the member is considered as misbehaving",
+			memberIndex,
+		)
+	}
+
+	// Construct a new view of the operating members according to the accepted
+	// DKG result.
+	operatingMemberIDs := make([]group.MemberIndex, 0)
```

Comment:
s/operatingMemberIDs/operatingMemberIndexes

Reply:
Done in 54916c43296c6911d2229b5ad1af34509c3d77cf.